### PR TITLE
Use bazel-contrib/setup-bazel instead of manually installing a deb during CI

### PR DESCRIPTION
**Pull request by dougthor42**
Use [`bazel-contrib/setup-bazel`](https://github.com/bazel-contrib/setup-bazel) instead of manually installing a deb during CI

Also fix some EOL whitespace because that's how my vim is configured.

@95-martin-orion

### CODE REVIEW

File: .github/workflows/bazeltest.yml [hunk 1]
```diff
@@ -22,12 +22,14 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+      - uses: bazel-contrib/setup-bazel@0.8.5
```
Comment (95-martin-orion) [inline .github/workflows/bazeltest.yml:25]: It's not immediately clear to me how (or if) this specifies the `bazel` version to install. IIRC, this is important for qsim as it needs to align with the bazel version used in [TFQ](https://github.com/tensorflow/quantum) for compatibility reasons.

Can we lock the bazel version to `5.3.0` here?

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]

File: .github/workflows/bazeltest.yml (hunk 1 above)
Comment (dougthor42) [reply .github/workflows/bazeltest.yml:25]: The bazel version is locked by `.bazelversion` in the root of the repo, and that's already set to `5.3.0`.

`setup-bazel` will install bazelisk 1.x as `bazel` and then any time `bazel` is called, bazelisk will download and install the bazel version defined by `.bazelversion` if the file exists (which it does :upside_down_face:)

More info on how bazelisk determines which version of bazel to run can be found here: https://github.com/bazelbuild/bazelisk?tab=readme-ov-file#how-does-bazelisk-know-which-bazel-version-to-run

**Comment by dougthor42** [comment]
BTW, the context for this is https://github.com/qh-lab/pyle/pull/46833#issuecomment-2204296035

**Review by 95-martin-orion — APPROVED** [review]
Changes LGTM, although it looks like we may need to claw back MacOS support before merging.

**Comment by 95-martin-orion** [comment]
@dougthor42 - #672 fixed the CI failures. If you're still interested in getting this merged, resolve conflicts and retry.

If I'm not available for final approval, request review from @pavoljuhas.

**Comment by dougthor42** [comment]
Conflicts fixed, please approve CI run.

**Pull request merged**